### ISSUE TITLE
Provisioner no longer expand $HOME on Terraform v1.1 or newer

### ISSUE THREAD

**Issue by jpablo-eqx** [equinix/terraform-metal-vsphere#52]
On Terraform v1.1 and above, the behavior of provisioner was change such that it no longer expand $HOME or ~: https://www.terraform.io/language/resources/provisioners/connection

```
╷
│ Error: file provisioner error
│
│   with null_resource.run_pre_reqs,
│   on main.tf line 177, in resource "null_resource" "run_pre_reqs":
│  177:   provisioner "file" {
│
│ Upload failed: scp: $HOME/bootstrap: No such file or directory
╵
```

https://github.com/equinix/terraform-metal-vsphere/blob/main/main.tf#L173

A workaround I found is to replace `$HOME` with `/root`. Doesn't affect terraform 1.0.0 or older.

**Comment by displague** [comment]
Thanks for identifying and surfacing this, @jpablo-eqx:

> Warning: In Terraform v1.0 and earlier, the built-in provisioners incorrectly passed the script_path value to scp through a remote shell and thus allowed it to be subject to arbitrary shell expansion, and thus created an unintended opportunity for remote code execution. Terraform v1.1 and later will now correctly quote and escape the script path to ensure that the remote scp process can always interpret it literally. For modules that will be used with Terraform v1.0 and earlier, avoid using untrusted external values as part of the script_path argument.

**Comment by displague** [comment]
Released as [v4.1.2](https://github.com/equinix/terraform-metal-vsphere/releases/tag/v4.1.2)
